guides/micronaut-data-jdbc-repository/kotlin/src/main/kotlin/example/micronaut/domain/Genre.kt

```diff
@@ -0,0 +1,13 @@
+package example.micronaut.domain
+
+import io.micronaut.data.annotation.GeneratedValue
+import io.micronaut.data.annotation.Id
+import io.micronaut.data.annotation.MappedEntity
+
+@MappedEntity
+data class Genre(
+    @field:Id
+    @field:GeneratedValue(GeneratedValue.Type.AUTO)
+    var id: Long? = null,
+    var name: String
```

Comment:
I think this would be more Kotlin idiomatic as:

```kotlin
@MappedEntity
data class Genre(val name: String) {
@GeneratedValue
@Id
var id: Long? = null
}
```

At least this is what we have in the data docs

Reply:
With Kotlin Data classes and `@MappedEntry` it needs `@field:` for `Id`. 
See: https://micronaut-projects.github.io/micronaut-data/latest/guide/#kotlinData
Note: that with the GeneratedValue type specified, `@field` is also needed.